### PR TITLE
Makes clown burgers honk

### PR DESCRIPTION
**Pull request by swindly**
:cl: Swindly
add: Clown burgers honk when eaten
balance: The maximum reagent volume of food decreases as it is eaten
/:cl:

[why]:  (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

I modified food's On_Consume so that it can do stuff when overridden. This provides a method of giving special effects to food as an alternative to adding reagents.

### CODE REVIEW

**Comment by Qustinnus** [comment]
use a squeeky component instead maybe

File: code/modules/food_and_drinks/food/snacks.dm [hunk 1]
```diff
@@ -36,8 +36,8 @@
 	else
 		..()
 
-/obj/item/reagent_containers/food/snacks/proc/On_Consume()
-	if(!usr)
+/obj/item/reagent_containers/food/snacks/proc/On_Consume(mob/M)
+	if(!usr || !M)
```
Comment (XDTM) [inline code/modules/food_and_drinks/food/snacks.dm]: I think you could just replace all instances of usr with M

**Comment by 81Denton** [comment]
Don't forget about Honkdaes

File: code/modules/food_and_drinks/food/snacks.dm (hunk 1 above)
Comment (swindly) [reply code/modules/food_and_drinks/food/snacks.dm]: They're different when someone feeds something to someone else.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]
use squeak component and add a on_consume signal registration

**Comment by Tlaltecuhtli** [comment]
why the balance tag?

**Comment by ninjanomnom** [comment]
because they used `balance:` in the changelog

**Pull request closed without merging**